dff/script/core/script.py

```diff
@@ -25,6 +26,80 @@
 
 
+def error_handler(error_msgs: list, msg: str, exception: Optional[Exception] = None):
+    """
+    This function handles errors during :py:class:`~dff.script.Script` validation.
+
+    :param error_msgs: List that contains error messages. :py:func:`~dff.script.error_handler`
+        adds every next error message to that list.
+    :param msg: Error message which is to be added into `error_msgs`.
+    :param exception: Invoked exception. If it has been set, it is used to obtain logging traceback.
+        Defaults to `None`.
+    :param logging_flag: The flag which defines whether logging is necessary. Defaults to `True`.
+    """
+    error_msgs.append(msg)
+    logger.error(msg, exc_info=exception)
+
+
+def validate_callable(callable: Callable, name: str, flow_label: str, node_label: str) -> List:
+    """
+    This function validates a function during :py:class:`~dff.script.Script` validation.
+    It checks parameter number (unconditionally), parameter types (if specified) and return type (if specified).
+
+    :param callable: Function to validate.
+    :param name: Type of the function (label, condition, response, etc.).
+    :param flow_label: Flow label this function is related to (used for error localization only).
+    :param node_label: Node label this function is related to (used for error localization only).
+    :param error_msgs: List that contains error messages. All the error message will be added to that list.
+    :param logging_flag: The flag which defines whether logging is necessary. Defaults to `True`.
+    :param expected_types: Tuple of types that correspond to expected function parameter types.
+        Also used for parameter number check. If `None`, parameter check is skipped. Defaults to `None`.
+    :param return_type: Tuple, containing expected function return type.
+        If `None` or contains more or less than 1 element, return type check is skipped. Defaults to `None`.
+    :return: list of produced error messages.
+    """
+    from dff.script.core.context import Context
+    from dff.pipeline.pipeline.pipeline import Pipeline
+
+    USER_FUNCTION_TYPES = {
+        "label": ((Context, Pipeline), None),
+        "response": ((Context, Pipeline), Message),
+        "condition": ((Context, Pipeline), bool),
+        "processing": ((Context, Pipeline), Context),
+    }
+
+    error_msgs = list()
+    signature = inspect.signature(callable)
+    function_type = name if name in USER_FUNCTION_TYPES.keys() else "processing"
+    arguments_type, return_type = USER_FUNCTION_TYPES[function_type]
+    params = list(signature.parameters.values())
+    if len(params) != len(arguments_type):
+        msg = (
+            f"Incorrect parameter number of {name}={callable.__name__}: "
+            f"should be {len(arguments_type)}, found {len(params)}, "
+            f"error was found in (flow_label, node_label)={(flow_label, node_label)}"
+        )
+        error_handler(error_msgs, msg, None)
+    if TYPE_CHECKING:
```

Comment:
Again, this won't work due to `TYPE_CHECKING` always being false at runtime.
Please, add tests for every possible error message during script validation that we do indeed raise that error.

Reply:
`TYPE_CHECKING` is not required in this function because you already import `Context` and `Pipeline` before defining `USER_FUNCTION_TYPES`. So there shouldn't be any problems if you simply remove the
`if TYPE_CHECKING` line in `validate_callable`.